tools/reactive-controllers/src/ColorController.ts

```diff
@@ -0,0 +1,520 @@
+                    g: string;
+                    b: string;
+                };
+                if (
+                    typeof h !== 'undefined' &&
+                    typeof s !== 'undefined' &&
+                    typeof l !== 'undefined'
+                ) {
+                    spaceId = 'hsl';
+                } else if (
+                    typeof h !== 'undefined' &&
+                    typeof s !== 'undefined' &&
+                    typeof v !== 'undefined'
+                ) {
+                    spaceId = 'hsv';
+                } else if (
+                    typeof r !== 'undefined' &&
+                    typeof g !== 'undefined' &&
+                    typeof b !== 'undefined'
+                ) {
+                    spaceId = 'srgb';
+                }
+            }
+        } else {
+            ({ spaceId } = this.color);
+        }
+        switch (spaceId) {
+            case 'hsv': {
+                const { h, s, v } = (this._color.to('hsv') as Color).hsv;
+                return {
+                    h,
+                    s: s / 100,
+                    v: v / 100,
+                    a: this._color.alpha,
+                };
+            }
+            case 'hsl': {
+                const { h, s, l } = (this._color.to('hsl') as Color).hsl;
+                return {
+                    h,
+                    s: s / 100,
+                    l: l / 100,
+                    a: this._color.alpha,
+                };
+            }
+            case 'srgb': {
+                const { r, g, b } = (this._color.to('srgb') as Color).srgb;
+                if (
+                    this._colorOrigin &&
+                    typeof (this._colorOrigin as { r: string }).r ===
+                        'string' &&
+                    (this._colorOrigin as { r: string }).r.search('%')
+                ) {
+                    return {
+                        r: `${Math.round(r * 255)}%`,
+                        g: `${Math.round(g * 255)}%`,
+                        b: `${Math.round(b * 255)}%`,
+                        a: this._color.alpha,
+                    };
+                }
+                return {
+                    r: Math.round(r * 255),
+                    g: Math.round(g * 255),
+                    b: Math.round(b * 255),
+                    a: this._color.alpha,
+                };
+            }
+        }
+        return this._color;
+    }
+
+    protected host: ReactiveElement;
+
+    get hue(): number {
+        return Number((this._color.to('hsl') as Color).hsl.h);
+    }
+
+    set hue(hue: number) {
+        this._color.set('h', hue);
+        this.host.requestUpdate();
+    }
+
+    constructor(
+        host: ReactiveElement,
+        {
+            manageAs,
+        }: {
+            manageAs?: string;
+        } = {}
+    ) {
+        this.host = host;
+        this.manageAs = manageAs;
+    }
+
+    getColor(format: string | ColorSpace): ColorObject {
```

Comment:
Can we have doc blocks for this and the subsequent methods?

Reply:
done!